include/swift/AST/TypeCheckRequests.h

```diff
@@ -1634,6 +1634,29 @@ class SynthesizeDefaultInitRequest
   bool isCached() const { return true; }
 };
 
+// SWIFT_ENABLE_TENSORFLOW
+class DifferentiableAttributeParameterIndicesRequest :
```

Comment:
Could this be named `DifferentiableAttributeTypeCheckRequest`? It currently does a lot more than calculate the parameter indices, so the current name is pretty confusing in the places where you're calling it to do something different. (Especially the accessordecls in SILFunctionBuilder.cpp).